data_node/upload_manager.go

```diff
@@ -102,8 +106,86 @@ func (um *UploadManager) handleInitialUpload(w http.ResponseWriter, r *http.Requ
 		return
 	}
 
-	w.WriteHeader(http.StatusCreated)
 	w.Header().Set("ID", id)
+	w.Header().Set("Max-Request-Size", strconv.FormatInt(um.maxChunkSize, 10))
+	w.WriteHeader(http.StatusCreated)
+}
+
+// handleAppendUpload is a function responsible for handling the first upload request
+func (um *UploadManager) handleAppendUpload(w http.ResponseWriter, r *http.Request) {
+	r.Body = http.MaxBytesReader(w, r.Body, um.maxChunkSize)
```

Comment:
Will this fail if the request is larger than the max value? Or will it truncate the request? I think we should fail the request altogether.

Reply:
The server will close connection and return

Reply:
Do you know what would be the status code or the error message? Can you try it?

Reply:
Looks like it was actually truncating the request, but it didn't close the connection, and it gets an error while trying to parse the body that closed the connection, so I added another checks